File: search-application/src/main/java/com/example/search/application/synonym/service/ListSynonymGroupsService.java
package com.example.search.application.synonym.service;

import com.example.search.application.synonym.port.in.ListSynonymGroupsUseCase;
import com.example.search.application.synonym.port.out.SynonymGroupRepository;
import com.example.search.domain.synonym.SynonymGroup;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ListSynonymGroupsService implements ListSynonymGroupsUseCase {

    private final SynonymGroupRepository repository;

    @Override
    @Transactional(readOnly = true)
    public List<SynonymGroup> listAll() {
        return repository.findAll();
    }
}

File: search-application/src/main/kotlin/com/example/search/application/synonym/service/ListSynonymGroupsService.kt
package com.example.search.application.synonym.service

import com.example.search.application.synonym.port.`in`.ListSynonymGroupsUseCase
import com.example.search.application.synonym.port.out.SynonymGroupRepository
import com.example.search.domain.synonym.SynonymGroup
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ListSynonymGroupsService(
    private val repository: SynonymGroupRepository
) : ListSynonymGroupsUseCase {

    @Transactional(readOnly = true)
    override fun listAll(): List<SynonymGroup> {
        return repository.findAll()
    }
}
